cpp/src/join/conditional_join.cu

```diff
@@ -49,7 +49,8 @@ std::unique_ptr<rmm::device_uvector<size_type>> conditional_join_anti_semi(
   if (right.num_rows() == 0) {
     switch (join_type) {
       case join_kind::LEFT_ANTI_JOIN:
-        return std::make_unique<rmm::device_uvector<size_type>>(left.num_rows(), stream, mr);
+        return get_trivial_left_join_indices(left, stream, rmm::mr::get_current_device_resource())
+          .first;
```

Comment:
This was returning an uninitialized vector, but it should be initialized with `[0, 1, 2, ..., left.num_rows() - 1]`.

Reply:
Why isn't the output generated using the provided `mr` but default mr?

Reply:
Good catch. Fixed in 1a0f66e0ffb52436caf2595ae0a2c140c4f186f1.